File: src/main/java/edu/kit/iti/formal/stvs/view/HostServiceSingleton.java
package edu.kit.iti.formal.stvs.view;

import javafx.application.HostServices;

/**
 * Created by csicar on 16.03.17.
 */
public class HostServiceSingleton {
  private static HostServices instance;

  public static void setInstance(HostServices newInstance) {
    if (instance != null) {
      throw new IllegalStateException("already set");
    }
    instance = newInstance;
  }

  public static HostServices getInstance() {
    return instance;
  }
}

File: stvs/src/main/kotlin/edu/kit/iti/formal/stvs/view/common/HostServiceSingleton.kt
package edu.kit.iti.formal.stvs.view.common

import javafx.application.HostServices

/**
 * Created by csicar on 16.03.17.
 */
object HostServiceSingleton {
    var instance: HostServices? = null
        set(newInstance) {
            check(field == null) { "already set" }
            field = newInstance
        }
}
